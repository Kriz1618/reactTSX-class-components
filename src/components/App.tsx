import React from 'react';
import { User } from '../model/Model';
import { AuthService, DataService } from '../services/';
import { Login } from './Login';
import { Router, Route, Switch } from 'react-router-dom';
import history from '../utils/history';
import { Navbar, Home, Profile, Spaces } from './';

interface AppState {
  user: User | undefined
}

/* istanbul ignore file */
export class App extends React.Component<{}, AppState> {

  private authService: AuthService = new AuthService();

  private dataService: DataService = new DataService();

  constructor(props: any) {
    super(props);
    this.state = { user: undefined };
    this.setUser = this.setUser.bind(this);
  }

  private setUser(user: User) {
    this.setState({ user: user });
  }

  render() {
    return <div className='wrapper'>
      <Router history={history}>
        <div>
          <Navbar user={this.state.user} />
          <Switch>
            <Route exact path='/' component={Home} />
            <Route exact path='/login'>
              <Login authService={this.authService} setUser={this.setUser} />
            </Route>
            <Route exact path='/profile'>
              <Profile user={this.state.user} authService={this.authService} />
            </Route>
            <Route exact path='/spaces'>
              <Spaces dataService={this.dataService} />
            </Route>
          </Switch>
        </div>
      </Router>
    </div >
  }
}
