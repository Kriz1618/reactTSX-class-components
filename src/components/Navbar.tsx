import React from 'react'
import { User } from '../model/Model'
import { Link } from 'react-router-dom'

export class Navbar extends React.Component<{ user: User | undefined }> {

  render() {
    let loginLogout: any;
    if (this.props.user) {
      loginLogout = <Link to='/logout' style={{ float: 'right' }}>{this.props.user.userName}</Link>
    } else {
      loginLogout = <Link data-testid='login-link' to='/login' style={{ float: 'right' }}>Login</Link>;
    }

    return (
      <div className="navbar">
        <Link data-testid='home-link' to='/'> Home</Link>
        <Link data-testid='profile-link' to='/profile'> Profile</Link>
        <Link data-testid='spaces-link' to='/spaces'> Spaces</Link>
        {loginLogout}
      </div>
    )

  }
}
