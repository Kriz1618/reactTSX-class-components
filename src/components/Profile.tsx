import React from 'react';
import { Link } from 'react-router-dom';
import { User, UserAttribute } from '../model/Model';
import { AuthService } from '../services/AuthService';

interface ProfileStates {
  userAttributes: UserAttribute[];
}

interface ProfileProps {
  user: User | undefined;
  authService: AuthService;
}

export class Profile extends React.Component<ProfileProps, ProfileStates> {
  state: ProfileStates = {
    userAttributes: []
  }

  async componentDidMount() {
    if (this.props.user) {
      const userAtrs = await this.props.authService.getUserAttributes(this.props.user);
      this.setState({ userAttributes: userAtrs })
    }
  }
  private renderUserAttributes() {
    const rows = [];
    for (const userAttribute of this.state.userAttributes) {
      rows.push(<tr key={userAttribute.Name}>
        <td>{userAttribute.Name}</td>
        <td>{userAttribute.Value}</td>
      </tr>)
    }

    return <table>
      <tbody>
        {rows}
      </tbody>
    </table>
  }
  render() {
    let profileSpace;
    if (this.props.user) {
      profileSpace = <div>
        <h3>Hello {this.props.user.userName}</h3>
        {this.renderUserAttributes()}
      </div>
    } else {
      profileSpace = <div>
        Please <Link to='/login'>Login</Link>
      </div>
    }

    return (
      <div>
        Profile page
        {profileSpace}
      </div>
    )

  }
}
