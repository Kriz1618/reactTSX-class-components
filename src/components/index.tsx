export { App } from "./App";
export { Home } from "./Home";
export { Login } from "./Login";
export { Navbar } from "./Navbar";
export { Profile } from "./Profile";
export { ConfirmModal } from "./spaces/ConfirmModal";
export { SpaceComponent } from "./spaces/SpaceComponent";
export { Spaces } from "./spaces/Spaces";
