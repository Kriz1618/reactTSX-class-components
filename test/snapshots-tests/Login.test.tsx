import { Login } from '../../src/components/Login'
import { create } from 'react-test-renderer';


describe('Login component snapshot testing', () => {

    test('initial test', () => {
        const snap = create(<Login
            authService={{} as any}
            setUser={{} as any}
        />)
        expect(snap).toMatchSnapshot();
    })

})