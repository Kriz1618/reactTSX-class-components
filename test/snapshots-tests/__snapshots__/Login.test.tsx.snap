// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component snapshot testing initial test 1`] = `
<div>
  <h2>
    Please Login
  </h2>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      value=""
    />
    <br />
    <input
      onChange={[Function]}
      type="password"
      value=""
    />
    <br />
    <input
      type="submit"
      value="Login"
    />
  </form>
</div>
`;
